uamqp/aio/_cbs_async.py

```diff
@@ -0,0 +1,218 @@
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#-------------------------------------------------------------------------
+
+import logging
+import asyncio
+from datetime import datetime
+
+from ._management_link_async import ManagementLink
+from ..utils import utc_now, utc_from_timestamp
+from ..message import Message, Properties
+from ..error import (
+    AuthenticationException,
+    TokenAuthFailure,
+    TokenExpired,
+    ErrorCondition
+)
+from ..constants import (
+    CbsState,
+    CbsAuthState,
+    CBS_PUT_TOKEN,
+    CBS_EXPIRATION,
+    CBS_NAME,
+    CBS_TYPE,
+    CBS_OPERATION,
+    ManagementExecuteOperationResult,
+    ManagementOpenResult,
+    DEFAULT_AUTH_TIMEOUT
+)
+from ..cbs import (
+    check_put_timeout_status,
+    check_expiration_and_refresh_status
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class CBSAuthenticator(object):
+    def __init__(
+        self,
+        session,
+        auth,
+        **kwargs
+    ):
+        self._session = session
+        self._connection = self._session._connection
+        self._mgmt_link = self._session.create_request_response_link_pair(
+            endpoint='$cbs',
+            on_amqp_management_open_complete=self._on_amqp_management_open_complete,
+            on_amqp_management_error=self._on_amqp_management_error,
+            status_code_field=b'status-code',
+            status_description_field=b'status-description'
+        )  # type: ManagementLink
+        self._auth = auth
+        self._encoding = 'UTF-8'
+        self._auth_timeout = kwargs.pop('auth_timeout', DEFAULT_AUTH_TIMEOUT)
+        self._token_put_time = None
+        self._expires_on = None
+        self._token = None
+        self._refresh_window = None
+
+        self._token_status_code = None
+        self._token_status_description = None
+
+        self.state = CbsState.CLOSED
+        self.auth_state = CbsAuthState.Idle
```

Comment:
Why is CbsState uppscase but not CbsAuthState? :D

Reply:
nice catch, I updated all the enum to be of upper cases